tests/s3_util_tests.c

```diff
@@ -144,6 +144,121 @@ static int s_test_s3_strip_quotes(struct aws_allocator *allocator, void *ctx) {
     return 0;
 }
 
+AWS_TEST_CASE(test_s3_parse_request_range_header, s_test_s3_parse_request_range_header)
+static int s_test_s3_parse_request_range_header(struct aws_allocator *allocator, void *ctx) {
+
+    (void)ctx;
+
+    struct range_header_example {
+        struct aws_byte_cursor header_value;
+
+        bool initial_message_has_start_range;
+        bool initial_message_has_end_range;
+        uint64_t initial_object_range_start;
+        uint64_t initial_object_range_end;
```

Comment:
why are the bools "initial_message" but the ints "initial_object"? Is there some subtle difference I'm not picking up on?

(personally, I would omit the "initial_X" prefix from all of them)